PSP-1711 automatic versioning

asanchezr opened the pull request:
Implements a new GitHub action to automatically bump the build version number for PRs merged to DEV. 
It uses consistent versioning for backend and frontend sub-projects.

The agreed version number format is:

```
{major}.{minor}.{patch/hotfix}-{ISnumber}.{build}
```

Fosol reviewed (approved):
Not sure I understand how this is working.  How do you control the version numbers?

devinleighsmith commented on build/bump-version.js:
```diff
@@ -0,0 +1,138 @@
+/**
+ * USAGE:
+ *
+ * node ./build/bump-version.js [<new-version> | major | minor | patch | is | build] [--apply] [--release-version]
```
I'd be tempted to put the purpose of the included options in this section, ie: what does release-version do?
I know because I read the code, but its nice to have the expected functionality documented here.

asanchezr replied:
I did put a comment on the next line saying what "--release-number" does LOL. I can make the note more prominent. Also maybe change the flag to "--print-version" might indicate what it does a bit better?

devinleighsmith commented:
I think the thing that I'm missing is what the plan would be for updating the IS and "resetting" the build version in Dev after a sprint. We could use your script to increment the IS numbers and push a commit for that, but then wouldn't your GH action run and increment the build number unnecessarily?

devinleighsmith commented:
Any chance of getting make integration for this as well ;-)?

asanchezr commented:
I have updated the files as per our chat in MS Teams. Fixed a bug and improved script documentation.

As stated [here](https://github.blog/changelog/2021-02-08-github-actions-skip-pull-request-and-push-workflows-with-skip-ci/) we can skip the auto-versioning action by including **[skip ci]** on the title of the PR that resets the version number on Dev after a sprint; e.g. 

```
build(release): Bump version to v1.0.0-10.0 [skip ci]
```
